Terminal.Gui/Views/ListView.cs

```diff
@@ -551,15 +551,17 @@ public class ListWrapper : IListDataSource {
 		/// <param name="source"></param>
 		public ListWrapper (IList source)
 		{
-			count = source.Count;
-			marks = new BitArray (count);
-			this.src = source;
+			if (source != null) {
+				count = source.Count;
+				marks = new BitArray (count);
+				this.src = source;
+			}
 		}
 
 		/// <summary>
 		/// Gets the number of items in the <see cref="IList"/>.
 		/// </summary>
-		public int Count => src.Count;
+		public int Count => Convert.ToInt32 (src?.Count);
```

Comment:
Why is the Convert.ToInt32 needed? Thanks.

Reply:
This was to handle src?.Count being null and therefore return 0

Reply:
Neat trick. The likelyhood is pretty low, but on a 64 bit system, this will be a bug with an insanely large `Count`, right?

Reply:
It's needed because the local IList src could be null.  Without this check it will crap out trying to read the Count property if a null collection was passed in the ctor.

Reply:
I understand the need for check in the case `src` is `null`. But I'm worried your shortcut of using `Convert` introduces a int64 to int32 dataloss bug.

Reply:
On 64 bit systems `.Count` is a 64 bit integer.

Reply:
Understand your now :)

Have just made a tweak to keep to the int type.  

You could even do it like this, but IMHO it's not nice to read..

```csharp
public int Count => src?.Count != null ? src.Count : 0;

Reply:
Uh, don't you mean:

```
public int Count => src != null ? src.Count : 0;
```

Reply:
both actually work - but that's more concise.

Is the backing field int count actually needed?  Couldn't this be replaced by using the property?